pandas/io/stata.py

```diff
@@ -1488,6 +1519,10 @@ def _read_strls(self) -> None:
             self.GSO[str(v_o)] = decoded_va
 
     def __next__(self) -> DataFrame:
+        if self._chunksize is None:
+            raise ValueError(
+                "chunksize must be set to a positive integer to use as an iterator."
+            )
```

Comment:
This breaks when `read_stata()` is passed `iterator=True` but no chunksize, #37280